packages/backend-core/src/users/utils.ts

```diff
@@ -67,21 +65,26 @@ export async function validateUniqueUser(email: string, tenantId: string) {
 }
 
 /**
- * For the given user id's, return the account holder if it is in the ids.
+ * For a list of users, return the account holder if there is an email match.
  */
-export async function getAccountHolderFromUserIds(
-  userIds: string[]
-): Promise<CloudAccount | undefined> {
-  if (!env.SELF_HOSTED && !env.DISABLE_ACCOUNT_PORTAL) {
-    const tenantId = getTenantId()
-    const account = await getAccountByTenantId(tenantId)
-    if (!account) {
-      throw new Error(`Account not found for tenantId=${tenantId}`)
-    }
-
-    const budibaseUserId = account.budibaseUserId
-    if (userIds.includes(budibaseUserId)) {
-      return account
+export async function getAccountHolderFromUsers(
+  users: Array<{
+    userId: string
+    email: string
+  }>
+): Promise<
+  | {
+      userId: string
+      email: string
     }
+  | undefined
+> {
+  if (!env.SELF_HOSTED && !env.DISABLE_ACCOUNT_PORTAL) {
+    const accountMetadata = await getExistingAccounts(
+      users.map(user => user.email)
+    )
+    return users.find(user =>
+      accountMetadata.map(metadata => metadata.email).includes(user.email)
```

Comment:
will `accountMetadata` be updated if an account holder changes their email address like in the v2 flows?

Reply:
It will have to be. Currently account metadata is created and deleted as part of the onboarding/offboarding flows. With this change, account metadata will be a core document that will need to be handled within transactions in our new flows.